### PR TITLE
fix: sdk7 avatar shape transform-tween reaction

### PR DESCRIPTION
**Pull request by pravusjif**
* Added needed `OnTransformChanged` event call on sdk7 Transform handler and Tween systems, so that the AvatarShape can reacto to the transform modifications and interpolate accordingly.
* Removed `in` modifier in some `OnTransformChanged()` methods as that is most certainly worsening performance since we don't pass strictly readonly structs as parameters (as explained in the paragraph before the conclusion of [this official blog post](https://devblogs.microsoft.com/premier-developer/the-in-modifier-and-the-readonly-structs-in-c/) regarding the `in` modifier)

Fixes https://github.com/decentraland/sdk/issues/1052

### <samp>🤖[[deprecated]](https://githubnext.com/copilot-for-prs-sunset) Generated by Copilot at 5dd740a</samp>

The pull request updates the `OnTransformChange` action in the `IDCLEntity` interface and its implementations to use `Vector3` and `Quaternion` parameters instead of an `object` parameter. This improves the code style, avoids unnecessary casting, and matches the ECS model. The pull request also adds calls to the `OnTransformChange` action in the `ECSTransformHandler` and `ECSTweenSystem` classes to trigger the interpolation logic for the `AvatarShape` component.

## Test instructions

* Avatar Tween rotation [playground link](https://playground.decentraland.org/?code=Y29uc3QgYXZhdGFyU2hhcGVFbnRpdHkgPSBlbmdpbmUuYWRkRW50aXR5KCkKVHJhbnNmb3JtLmNyZWF0ZShhdmF0YXJTaGFwZUVudGl0eSwgewogIHBvc2l0aW9uOiBWZWN0b3IzLmNyZWF0ZSg4LCAwLjI1LCA4KSwKfSkKClR3ZWVuLmNyZWF0ZShhdmF0YXJTaGFwZUVudGl0eSwgewogIGR1cmF0aW9uOiA4MDAwLAogIGVhc2luZ0Z1bmN0aW9uOiBFYXNpbmdGdW5jdGlvbi5FRl9MSU5FQVIsCiAgY3VycmVudFRpbWU6IDAsCiAgcGxheWluZzogdHJ1ZSwKICBtb2RlOiBUd2Vlbi5Nb2RlLlJvdGF0ZSh7CiAgICBzdGFydDogUXVhdGVybmlvbi5JZGVudGl0eSgpLAogICAgZW5kOiBRdWF0ZXJuaW9uLmZyb21FdWxlckRlZ3JlZXMoMCwgMTgwLCAwKQogIH0pCn0pCgpBdmF0YXJTaGFwZS5jcmVhdGUoYXZhdGFyU2hhcGVFbnRpdHksIHsKICB3ZWFyYWJsZXM6IFsKICAgICJ1cm46ZGVjZW50cmFsYW5kOm1hdGljOmNvbGxlY3Rpb25zLXYyOjB4ZWU3N2IwYTEwNGNkNmRiMWJiYmNmYTJmMTMwNzZmMjM0NjQ3YzAxNzoyIiwKICAgICJ1cm46ZGVjZW50cmFsYW5kOm1hdGljOmNvbGxlY3Rpb25zLXYyOjB4NzQyOWM4OGUzMzI4ODU5OWRmYTVkZWZiN2VhZTQ5YWU5NWFkNDRjNTowIiwKICAgICJ1cm46ZGVjZW50cmFsYW5kOm1hdGljOmNvbGxlY3Rpb25zLXYyOjB4M2E1M2FmY2Q0ZjNhNDA5NTNmYTEyMTdhNTYyNjU5MDliYjJmNjMwOTozIiwKICAgICJ1cm46ZGVjZW50cmFsYW5kOm1hdGljOmNvbGxlY3Rpb25zLXYyOjB4M2E1M2FmY2Q0ZjNhNDA5NTNmYTEyMTdhNTYyNjU5MDliYjJmNjMwOTowIiwKICAgICJ1cm46ZGVjZW50cmFsYW5kOm1hdGljOmNvbGxlY3Rpb25zLXYyOjB4MzZlOTRkMTI4OTkzMTdkYzc5MmU1NThmOTkzMjEzZWVmOWI4NWM3ODowIiwKICAgICJ1cm46ZGVjZW50cmFsYW5kOm1hdGljOmNvbGxlY3Rpb25zLXYyOjB4ZmY4ZTYzMGJkNDMyNDZiZjVjMzdhZWNjMDIyOGFmNTQxOGE4Y2VlZjo0IgogIF0sCiAgaWQ6ICJ0ZXN0LWF2YXRhciIsCiAgZW1vdGVzOiBbXQp9KQ%3D%3D&explorer-branch=fix/sdk7-avatar-shape-transform-and-tween-reaction): after the scene loads hit the RUN button and confirm that the NPC avatar is rotated.
* Avatar Transform rotation [playground link](https://playground.decentraland.org/?code=Y29uc3QgYXZhdGFyU2hhcGVFbnRpdHkgPSBlbmdpbmUuYWRkRW50aXR5KCkKVHJhbnNmb3JtLmNyZWF0ZShhdmF0YXJTaGFwZUVudGl0eSwgewogIHBvc2l0aW9uOiBWZWN0b3IzLmNyZWF0ZSg4LCAwLjI1LCA4KSwKfSkKCkF2YXRhclNoYXBlLmNyZWF0ZShhdmF0YXJTaGFwZUVudGl0eSwgewogIHdlYXJhYmxlczogWwogICAgInVybjpkZWNlbnRyYWxhbmQ6bWF0aWM6Y29sbGVjdGlvbnMtdjI6MHhlZTc3YjBhMTA0Y2Q2ZGIxYmJiY2ZhMmYxMzA3NmYyMzQ2NDdjMDE3OjIiLAogICAgInVybjpkZWNlbnRyYWxhbmQ6bWF0aWM6Y29sbGVjdGlvbnMtdjI6MHg3NDI5Yzg4ZTMzMjg4NTk5ZGZhNWRlZmI3ZWFlNDlhZTk1YWQ0NGM1OjAiLAogICAgInVybjpkZWNlbnRyYWxhbmQ6bWF0aWM6Y29sbGVjdGlvbnMtdjI6MHgzYTUzYWZjZDRmM2E0MDk1M2ZhMTIxN2E1NjI2NTkwOWJiMmY2MzA5OjMiLAogICAgInVybjpkZWNlbnRyYWxhbmQ6bWF0aWM6Y29sbGVjdGlvbnMtdjI6MHgzYTUzYWZjZDRmM2E0MDk1M2ZhMTIxN2E1NjI2NTkwOWJiMmY2MzA5OjAiLAogICAgInVybjpkZWNlbnRyYWxhbmQ6bWF0aWM6Y29sbGVjdGlvbnMtdjI6MHgzNmU5NGQxMjg5OTMxN2RjNzkyZTU1OGY5OTMyMTNlZWY5Yjg1Yzc4OjAiLAogICAgInVybjpkZWNlbnRyYWxhbmQ6bWF0aWM6Y29sbGVjdGlvbnMtdjI6MHhmZjhlNjMwYmQ0MzI0NmJmNWMzN2FlY2MwMjI4YWY1NDE4YThjZWVmOjQiCiAgXSwKICBpZDogInRlc3QtYXZhdGFyIiwKICBlbW90ZXM6IFtdCn0pCgplbmdpbmUuYWRkU3lzdGVtKChkdCkgPT4gewogIGNvbnN0IGF2YXRhclRyYW5zZm9ybSA9IFRyYW5zZm9ybS5nZXRNdXRhYmxlKGF2YXRhclNoYXBlRW50aXR5KQoKICBhdmF0YXJUcmFuc2Zvcm0ucm90YXRpb24gPSBRdWF0ZXJuaW9uLnJvdGF0ZVRvd2FyZHMoYXZhdGFyVHJhbnNmb3JtLnJvdGF0aW9uLCBRdWF0ZXJuaW9uLmZyb21FdWxlckRlZ3JlZXMoMCwgMTgwLCAwKSwgMC41KQp9KQ%3D%3D&explorer-branch=fix/sdk7-avatar-shape-transform-and-tween-reaction): after the scene loads hit the RUN button and confirm that the NPC avatar is rotated.

The RUN button can be used more than once, to restart the scene.

### CODE REVIEW

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Handler/ECSTransformHandler.cs [hunk 1]
```diff
@@ -92,6 +92,11 @@ public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, ECSTr
 
             if (scaleChange || rotationChange)
                 sbcInternalComponent.OnTransformScaleRotationChanged(scene, entity);
+
+            // Same AvatarShape interpolation used at DCLTransform from SDK6
+            // We only want to trigger it if there's a method subscribed
+            if (entity.OnTransformChange != null)
+                entity.OnTransformChange.Invoke(model.position, model.rotation);
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Handler/ECSTransformHandler.cs:99]: just some sugar  - `entity.OnTransformChange?.Invoke(model.position, model.rotation);`

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs [hunk 2]
```diff
@@ -94,6 +94,11 @@ private void UpdateTweenComponentModel(KeyValueSetTriplet<IParcelScene, long, EC
 
             model.currentTime = currentTime;
             tweenInternalComponent.PutFor(scene, entityId, model);
+
+            // Same AvatarShape interpolation used at DCLTransform from SDK6
+            // We only want to trigger it if there's a method subscribed
+            if (entity.OnTransformChange != null)
+                entity.OnTransformChange.Invoke(model.transform.localPosition, model.transform.localRotation);
```
Comment (popuz) [inline unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs:101]: another sugar - `entity.OnTransformChange?.Invoke(model.transform.localPosition, model.transform.localRotation);`

**Review by popuz — APPROVED** [review]
LGTM 🦅 And thanks for interesting reading ;)  (I left 2 synaxis sugar comments)

**Review by anicalbano — APPROVED** [review]
✔️ Tested and approved by QA

https://github.com/decentraland/unity-renderer/assets/139157886/4b44e0fe-481e-4e8c-bb79-ca00a9814746

https://github.com/decentraland/unity-renderer/assets/139157886/b9f1389f-d18f-4504-b32e-40d10e26d171

File: unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Handler/ECSTransformHandler.cs (hunk 1 above)
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/Transform/Handler/ECSTransformHandler.cs]: Totally agreee, I didn't notice that, thanks!

File: unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs (hunk 2 above)
Comment (pravusjif) [reply unity-renderer/Assets/DCLPlugins/ECS7/Systems/TweenSystem/ECSTweenSystem.cs]: Done!

**Pull request merged**